### PR TITLE
BaSD-MS-60: Postulants List and Form

### PR DESCRIPTION
**Pull request by ortizjoaquin**
Create POSTULANT form screen ✅
Create POSTULANT list screen ✅
Tested and working fine on local host ✅

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,8 @@
     "@testing-library/jest-dom": "^5.15.1",
     "@testing-library/react": "^11.2.7",
     "@testing-library/user-event": "^12.8.3",
+    "express": "^4.17.1",
+    "mongoose": "^6.0.13",
```
Comment (estefrare) [inline package.json]: Why did you install express and mongoose in the app?

File: src/Components/Postulants/Form/index.js [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+  const [workExperienceEDValue, setWorkExperienceEDValue] = useState('');
+  const [workExperienceDescriptionValue, setWorkExperienceDescriptionValue] = useState('');
+
+  if (postulantId) {
+    useEffect(() => {
+      fetch(`${process.env.REACT_APP_API}/postulants?_id=${postulantId}`)
+        .then((response) => response.json())
+        .then((response) => {
+          autoFill(response);
+        });
```
Comment (estefrare) [inline src/Components/Postulants/Form/index.js:52]: add `.catch` and handle the errors

File: src/Components/Postulants/Form/index.js [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+            throw new Error(message);
+          });
+        }
+        return response.json();
+      })
+      .then(() => {
+        window.location.href = `${window.location.origin}/postulants`;
+      })
+      .catch((error) => {
+        console.log(JSON.stringify(error));
```
Comment (estefrare) [inline src/Components/Postulants/Form/index.js]: the user will never know what happened if we do not show it in the UI

File: src/Components/Postulants/Form/index.js [hunk 4]
```diff
@@ -0,0 +1,714 @@
...
+    const fillSecStudy = data.data[0].studies.secondaryStudies;
+    const fillTerStudy = data.data[0].studies.tertiaryStudies[0];
+    const fillUniStudies = data.data[0].studies.universityStudies[0];
+    const fillInfStudies = data.data[0].studies.informalStudies[0];
+    const fillWorkExp = data.data[0].workExperience[0];
+
+    let from = fillData.contactRange?.from.toString();
+    let to = fillData.contactRange?.to.toString();
+    from = `${from.slice(0, 2)}:${from.slice(2)}`;
+    to = `${to.slice(0, 2)}:${to.slice(2)}`;
```
Comment (estefrare) [inline src/Components/Postulants/Form/index.js]: update the Schema to make it easy, just use the format that the time input return

File: src/Components/Postulants/index.js [hunk 5]
```diff
@@ -1,9 +1,101 @@
...
+  const [showModal, setShowModal] = useState(false);
+  const [postulants, setPostulants] = useState([]);
+  const [itemOnDelete, setItemOnDelete] = useState({});
+
+  useEffect(() => {
+    fetch(`${process.env.REACT_APP_API}/postulants`)
+      .then((response) => response.json())
+      .then((response) => {
+        setPostulants(response.data);
+      });
```
Comment (estefrare) [inline src/Components/Postulants/index.js:19]: add a `.catch()`

File: src/Components/Postulants/index.js [hunk 6]
```diff
@@ -1,9 +1,101 @@
...
+  const deletePostulant = (id) => {
+    const url = `${process.env.REACT_APP_API}/postulants/${id}`;
+    fetch(url, {
+      method: 'DELETE',
+      headers: {
+        'Content-type': 'application/json'
+      }
+    }).then(() => {
+      window.location.href = `${window.location.origin}/postulants`;
+    });
```
Comment (estefrare) [inline src/Components/Postulants/index.js]: add a `.catch()`

**Review by estefrare — CHANGES_REQUESTED** [review]

File: src/Components/Postulants/postulants.module.css [hunk 7]
```diff
@@ -1,6 +1,47 @@
 .container {
-  flex: 1;
   display: flex;
+  flex-direction: column;
   justify-content: center;
   align-items: center;
+}
+
+a {
```
Comment (AgustinCarthery) [inline src/Components/Postulants/postulants.module.css]: avoid using tag name selectors

File: package.json (hunk 1 above)
Comment (ortizjoaquin) [reply package.json]: Uninstalled both.

File: src/Components/Postulants/Form/index.js (hunk 2 above)
Comment (ortizjoaquin) [reply src/Components/Postulants/Form/index.js:52]: added `.catch()`

File: src/Components/Postulants/Form/index.js (hunk 3 above)
Comment (ortizjoaquin) [reply src/Components/Postulants/Form/index.js]: added `.catch()`

File: src/Components/Postulants/index.js (hunk 5 above)
Comment (ortizjoaquin) [reply src/Components/Postulants/index.js:19]: added `.catch()`

File: src/Components/Postulants/index.js (hunk 6 above)
Comment (ortizjoaquin) [reply src/Components/Postulants/index.js]: Added `.catch()`

File: src/Components/Postulants/Form/index.js (hunk 4 above)
Comment (ortizjoaquin) [reply src/Components/Postulants/Form/index.js]: added RegExp and now hours are stored as 'string' in mongoose

File: src/Components/Postulants/postulants.module.css (hunk 7 above)
Comment (ortizjoaquin) [reply src/Components/Postulants/postulants.module.css]: added classes to avoid most tag name selectors

**Review by estefrare — APPROVED** [review]

**Review by AgustinCarthery — APPROVED** [review]

**Pull request merged**